### PR TITLE
fix error from RVS2

### PR DESCRIPTION
**Pull request by jjhesk**
added json parsing
added listening on error #36

### CODE REVIEW

File: lib/ws.js [hunk 1]
```diff
@@ -29,16 +29,22 @@ class BinanceWS {
...
+              }
+               eventHandler(response);
+            }catch (e){
+                console.log("error from parsing body..");
             }
-
-            eventHandler(response);
+        });
+        ws.on('error', (erro) => {
+           console.log(erro);
```
Comment (aarongarvey) [inline lib/ws.js]: Can you remove this line and replace it with

`// node.js EventEmitters will throw and exit if no error listener is registered`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "binance",
-  "version": "1.3.1",
+  "version": "1.3.5",
```
Comment (aarongarvey) [inline package.json]: Why 5?  2 please

File: lib/ws.js (hunk 1 above)
Comment (jjhesk) [reply lib/ws.js]: ws on error was the exact solution to the bug i reported previously. i tested many times.

File: lib/ws.js [hunk 3]
```diff
@@ -29,16 +29,22 @@ class BinanceWS {
...
         const ws = new WebSocket(path);
         ws.on('message', (json) => {
-            let response = JSON.parse(json);
-            if (this._beautify) {
-                if (response.stream) {
-                    response.data = this._beautifyResponse(response.data);
-                } else {
-                    response = this._beautifyResponse(response);
-                }
+            try {
```
[the remark this answers is not in the document]
Comment (jjhesk) [reply lib/ws.js]: when the json is corrupted for some reasons, there is no point to report to the handler and let it run further. i think just skip that until the correct one is parsed next time.

**Comment by aarongarvey** [comment]
I went ahead and made the changes

**Comment by jjhesk** [comment]
thanks

**Pull request closed without merging**